code/edgeidentity/src/main/java/com/adobe/marketing/mobile/edge/identity/URLUtils.java

```diff
@@ -0,0 +1,108 @@
+/*
+  Copyright 2022 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.edge.identity;
+
+import com.adobe.marketing.mobile.LoggingMode;
+import com.adobe.marketing.mobile.MobileCore;
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.nio.charset.StandardCharsets;
+
+public class URLUtils {
+
+	static final String LOG_TAG = "URLUtils";
+
+	/**
+	 * Helper function to generate url variables in format acceptable by the AEP web SDKs
+	 *
+	 * @param ts timestamp {@link String} denoting time when url variables request was made
+	 * @param ecid Experience Cloud identifier {@link String} generated by the SDK
+	 * @param orgId Experience Cloud Org identifier {@link String} set in the configuration
+	 * @return {@link String} formatted with the visitor id payload
+	 */
+	static String generateURLVariablesPayload(final String ts, final String ecid, final String orgId) {
+		final StringBuilder urlFragment = new StringBuilder();
+
+		// construct the adobe_mc string
+
+		// append timestamp
+		String theIdString = appendKVPToVisitorIdString(null, IdentityConstants.UrlKeys.TS, ts);
+
+		// append ecid
+		theIdString = appendKVPToVisitorIdString(theIdString, IdentityConstants.UrlKeys.EXPERIENCE_CLOUD_ID, ecid);
+
+		// add Experience Cloud Org ID
+		theIdString = appendKVPToVisitorIdString(theIdString, IdentityConstants.UrlKeys.EXPERIENCE_CLOUD_ORG_ID, orgId);
+
+		// after the adobe_mc string is created, encode the idString before adding it to the url
+		urlFragment.append(IdentityConstants.UrlKeys.PAYLOAD);
+		urlFragment.append("=");
+
+		urlFragment.append(urlEncode(theIdString));
```

Comment:
Can we remove urlEncode method and use URLEncoder.encode in place and also remove all the ulrEncode tests? That should also help as a reminder that in case more advanced encoding is needed we can add additional helpers.